modules/autotagging-commons/common/src/main/java/org/opensearch/rule/FeatureValueMatch.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rule;
+
+/**
+ * Represents a feature value along with a matching score.
+ */
+public class FeatureValueMatch<V> {
```

Comment:
How about renaming this to just `Match` or `MatchLabel` since current name is bit too verbose and doesn't add any context